components/infra-proxy-service/server/automate_infra_server_users.go

```diff
@@ -100,3 +107,89 @@ func fromStorageToListAutomateInfraOrgUsers(ul []storage.OrgUser) []*response.Au
 
 	return tl
 }
+
+// ResetInfraServerUserKey updates the public key on the Chef Server and returns the private key
+func (s *Server) ResetInfraServerUserKey(ctx context.Context, req *request.ResetInfraServerUserKeyReq) (*response.ResetInfraServerUserKeyRes, error) {
+	err := validation.New(validation.Options{
+		Target:  "user",
+		Request: *req,
+		Rules: validation.Rules{
+			"ServerId": []string{"required"},
+			"Name":     []string{"required"},
+		},
+	}).Validate()
+
+	if err != nil {
+		log.Error("failed to validate: ", err)
+		return nil, err
+	}
+
+	server, err := s.service.Storage.GetServer(ctx, req.ServerId)
+	if err != nil {
+		log.Error("cannot get the server: ", err)
+		return nil, err
+	}
+	if server.CredentialID == "" {
+		return nil, errors.New("webui key is not available with server")
+	}
+	// Get web ui key from secrets service
+	secret, err := s.service.Secrets.Read(ctx, &secrets.Id{Id: server.CredentialID})
+	if err != nil {
+		log.Error("cannot read the secret: ", err)
+		return nil, err
+	}
+	c, err := s.createChefServerClient(ctx, req.ServerId, GetAdminKeyFrom(secret), "pivotal", true)
+	if err != nil {
+		log.Error("cannot create a client: ", err)
+		return nil, err
+	}
+
+	// Deletes the existing key
+	_, err = c.client.Users.DeleteKey(req.UserName, "default")
+	if err != nil {
+		log.Error("cannot delete the user's key: ", err)
+		return nil, ParseAPIError(err)
+	}
+	chefError, _ := chef.ChefError(err)
+	if err != nil && chefError.StatusCode() != 404 {
```

Comment:
we are handling the same error twice, if err is null why it will check for chef error?